tap-agent/src/metrics.rs

```diff
@@ -34,7 +34,8 @@ async fn _run_server(port: u16) {
         .route("/metrics", get(handler_metrics))
         .fallback(handler_404);
     let addr = SocketAddr::from(([0, 0, 0, 0], port));
-    let server = Server::bind(&addr).serve(app.into_make_service());
+    let listener = tokio::net::TcpListener::bind(addr).await.unwrap();
```

Comment:
use `expect` instead of `unwrap` and add information: "Failed to bind metrics address"